libtsuba/src/RDGCore.cpp

```diff
@@ -26,11 +26,29 @@ AddProperties(
     int last = current->num_columns();
 
     for (int i = 0, n = schema->num_fields(); i < n; i++) {
-      auto result =
-          next->AddColumn(last + i, schema->field(i), props->column(i));
+      auto name = schema->field(i)->name();
+      auto current_col = next->GetColumnByName(name);
+      arrow::Result<std::shared_ptr<arrow::Table>> result;
+      std::string error_context = "insert";
```

Comment:
Why isn't this defined (not declared) in the `if` condition?

Reply:
Well, I would declare it as an empty string and define it in both arms of the if.  I don't like doing that because if something gets screwed up, an empty string is a valid and confusing value.  So I like to initialize the string into a valid state.  I can then either reassign it to the same value in the upper if clause, which looks like an error when I read the code, or I can not assign it, which might look like an oversight.  So there is no perfect solution in my book.  That said, I'm happy to change this to include two assignments of "insert".

Reply:
Valid reason. It's fine as is.

Reply:
Added a comment.  Why not.